input/system/crunchy_bridge/system.go

```diff
@@ -0,0 +1,62 @@
+package crunchy_bridge
+
+import (
+	"time"
+
+	"github.com/pganalyze/collector/config"
+	"github.com/pganalyze/collector/input/system/selfhosted"
+	"github.com/pganalyze/collector/state"
+	"github.com/pganalyze/collector/util"
+)
+
+// GetSystemState - Gets system information about a Crunchy Bridge instance
+func GetSystemState(config config.ServerConfig, logger *util.Logger) (system state.SystemState) {
+	// With Crunchy Bridge, we are assuming that the collector is deployed on the Container Apps
+	// Most of the metrics can be obtained using the same way as the selfhosted
+	// (as the collector runs on the database server), except disk usage metrics
+	system = selfhosted.GetSystemState(config, logger)
+	system.Info.Type = state.CrunchyBridgeSystem
+
+	// When API key is provided, use API to obtain extra info including disk usage metrics
+	if config.CrunchyBridgeAPIKey == "" {
+		return
+	}
+	client := Client{Client: *config.HTTPClientWithRetry, BaseURL: apiBaseURL, BearerToken: config.CrunchyBridgeAPIKey, ClusterID: config.CrunchyBridgeClusterID}
+
+	clusterInfo, err := client.GetClusterInfo()
+	if err != nil {
+		logger.PrintError("CrunchyBridge/System: Encountered error when getting a cluster info %v\n", err)
+		return
+	}
+	system.Info.CrunchyBridge = &state.SystemInfoCrunchyBridge{
+		ClusterName: clusterInfo.Name,
+		PlanID:      clusterInfo.PlanID,
+		ProviderID:  clusterInfo.ProviderID,
+		RegionID:    clusterInfo.RegionID,
+		CPUUnits:    clusterInfo.CPU,
+		StorageGB:   clusterInfo.Storage,
+		MemoryGB:    clusterInfo.Memory,
+	}
+	if parsedCreatedAt, err := time.Parse(time.RFC3339, clusterInfo.CreatedAt); err != nil {
+		system.Info.CrunchyBridge.CreatedAt = parsedCreatedAt
+	}
+
+	diskUsageMetrics, err := client.GetDiskUsageMetrics()
+	if err != nil {
+		logger.PrintError("CrunchyBridge/System: Encountered error when getting cluster disk usage metrics %v\n", err)
+		return
+	}
+
+	system.DataDirectoryPartition = "/"
+	system.DiskPartitions = make(state.DiskPartitionMap)
+	// Manually specify the disk name to "md0" as that's the main disk
```

Comment:
[note for reviewer] We pass all the disk stats to pganalyze, and on the pganalyze side, we only persist the disk stat of the disk that matches to the data directory partition.
In this case, with `system.DataDirectoryPartition = "/"`, this is the data dir with the disk name `"md0"`. Within transformation, we then assign the index of `"md0"` disk as `DiskIdx`.
https://github.com/pganalyze/collector/blob/def695671efc4670f80a1867238177bbfcc36f60/output/transform/system.go#L278

Crunchy Bridge instance has several disks, and if we don't specify this part well, `0` will be used as `DiskIdx` and we'll incorrectly get the disk stats of `loop0` (which is Container Apps disk).

```
[13] pry[stag](main)> full.system.disk_references
=> [<Pganalyze::Collector::DiskReference: device_name: "loop0">,
 <Pganalyze::Collector::DiskReference: device_name: "md0">,
 <Pganalyze::Collector::DiskReference: device_name: "nvme0n1">,
 <Pganalyze::Collector::DiskReference: device_name: "nvme0n1p1">,
 <Pganalyze::Collector::DiskReference: device_name: "nvme0n1p2">,
 <Pganalyze::Collector::DiskReference: device_name: "nvme1n1">,
 <Pganalyze::Collector::DiskReference: device_name: "nvme2n1">,
 <Pganalyze::Collector::DiskReference: device_name: "nvme3n1">,
 <Pganalyze::Collector::DiskReference: device_name: "nvme4n1">]
 ```